src/expense.ts

```diff
@@ -0,0 +1,98 @@
+import { collection, doc, addDoc, deleteDoc, updateDoc, Timestamp } from 'firebase/firestore';
+
+import db from './firebase';
+
+export interface ExpenseDataFirestore {
+  title: string;
+  date: Timestamp;
+  amount: number;
+}
+
+export interface ExpenseDataForm {
+  title: string;
+  date: Date;
+  amount: string;
+}
+
+export class Expense {
+  title: string;
+  date: Date;
+  amount: number;
+  id: string;
+
+  constructor(expense: ExpenseDataFirestore, id: string) {
```

Comment:
maybe add a little class description about the two ways how to instantiate an expense (constructor/existing in the db vs. `add`)